src/stan_math_backend/Transform_Mir.ml

```diff
@@ -240,24 +305,72 @@ let trans_prog (p : typed_prog) =
         true
     | _ -> false
   in
+  let translate_to_open_cl stmts =
+    if use_opencl then
+      let data_var_idents = List.map ~f:fst p.input_vars in
+      let rec trans_stmt_to_opencl s =
+        { s with
+          stmt=
+            map_statement
+              (switch_expr_to_opencl data_var_idents)
+              trans_stmt_to_opencl s.stmt }
+      in
+      List.map stmts ~f:trans_stmt_to_opencl
+    else []
```

Comment:
Is this the OCaml way of doing this or is there a better way? Teach me :)

Reply:
Doesn't this return an empty list if the flag isn't set? Which would eat all the statements in the pipeline right? I think this should return `stmts` in the `else` case probs?